easybuild/easyblocks/w/wien2k.py

```diff
@@ -101,21 +101,27 @@ def configure_step(self):
 
         # toolchain-dependent values
         comp_answer = None
-        if self.toolchain.comp_family() == toolchain.INTELCOMP:  #@UndefinedVariable
+        if self.toolchain.comp_family() == toolchain.INTELCOMP:  # @UndefinedVariable
             if LooseVersion(get_software_version("icc")) >= LooseVersion("2011"):
-                comp_answer = 'I'  # Linux (Intel ifort 12.0 compiler + mkl )
+                if LooseVersion(self.version) < LooseVersion("17"):
+                    comp_answer = 'I'  # Linux (Intel ifort 12.0 compiler + mkl )
+                else:
+                    comp_answer = 'LI'  # Linux (Intel ifort compiler (12.0 or later)+mkl+intelmpi))
             else:
                 comp_answer = "K1"  # Linux (Intel ifort 11.1 compiler + mkl )
-
-        elif self.toolchain.comp_family() == toolchain.GCC:  #@UndefinedVariable
-            comp_answer = 'V'  # Linux (gfortran compiler + gotolib)
-
+        elif self.toolchain.comp_family() == toolchain.GCC:  # @UndefinedVariable
+            if LooseVersion(self.version) < LooseVersion("17"):
+                comp_answer = 'V'  # Linux (gfortran compiler + gotolib)
+            else:
+                comp_answer = 'LG'  # Linux (gfortran compiler + OpenBlas)
         else:
             raise EasyBuildError("Failed to determine toolchain-dependent answers.")
 
         # libraries
-        rlibs = "%s %s" % (os.getenv('LIBLAPACK_MT'), self.toolchain.get_flag('openmp'))
-        rplibs = [os.getenv('LIBSCALAPACK_MT'), os.getenv('LIBLAPACK_MT')]
+        liblapack = os.getenv('LIBLAPACK_MT').replace('static', 'dynamic')
+        libscalapack = os.getenv('LIBSCALAPACK_MT').replace('static', 'dynamic')
```

Comment:
@migueldiascosta Can you explain this change? Maybe include a comment above these two lines to explain why you're doing this?

Reply:
well, the new `siteconfig_lapw` script insists on adding the dynamic `mkl_scalapack` instead of using `RP_LIBS`. at least in `lapw2_mpi`, switching everything to dynamic solved the problem, but I suppose we can also patch the script...

Reply:
They probably changed that for a reason?

Just wondering whether we should also do this for old WIEN2k versions...

Reply:
I don't think they had a specific reason. Also, I tested the block with `14.2` and `16.1` and it still works.

But if we really want to use static libraries for some other reason, I can try harder to make it work